python/ray/tests/test_advanced_3.py

```diff
@@ -886,6 +886,64 @@ def g():
     assert ray.get(f.remote()) is None
 
 
+def test_override_environment_variables_env_caching(shutdown_only):
+    """Test that workers with specified envs are cached and reused."""
+    ray.init()
+
+    env_var_name = "TEST123"
+    val1 = "VAL1"
+    val2 = "VAL2"
+    assert os.environ.get(env_var_name) is None
+
+    def task():
+        return os.environ.get(env_var_name), os.getpid()
+
+    @ray.remote
+    def f():
+        return task()
+
+    @ray.remote
+    def g():
+        return task()
+
+    assert ray.get(f.remote())[0] is None
```

Comment:
This test is a bit confusing (and I don't know what it tries doing). Can you add some comments to explain more details? Is it that the worker is reused for the same env variable?

Reply:
Yes, it's for testing that the same worker is reused when the runtime env is the same--let me add some comments